### PR TITLE
Editor dnd

### PR DESCRIPTION
**Pull request by kodeebo**
Create slate plugin to handle dnd

- use native html event handlers and "draggable" attribute to make embeds and boxes draggable with thumbnails
- Text dragging is just handled "natively", unless it is a list or table and more than one cell / item is selected, then the whole table / list is moved

Check that:
- [x] Everything can be dragged more or less
- [x] It is moved before or after content based on drop distance

### CODE REVIEW

File: src/components/SlateEditor/plugins/DND/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  event.stopPropagation();
+  event.preventDefault();
+}
+
+const shouldCopyTableOrList = (type, editor) => {
+  if (type === 'table' || type.includes('list')) {
+    const nativeSelection = window.getSelection();
+    const range = findRange(nativeSelection, editor);
+    const nodesInRange = editor.value.document.getLeafBlocksAtRange(range);
+    if (nodesInRange.size > 1) return true;
```
Comment (chrpeter) [inline src/components/SlateEditor/plugins/DND/index.js]: return nodesInRange.size > 1 ? og curly brackets

File: src/components/SlateEditor/plugins/DND/utils.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const getTopNode = (node, editor) => {
+  const parent = editor.value.document.getParent(node.key);
+  if (!parent) return null;
```
Comment (chrpeter) [inline src/components/SlateEditor/plugins/DND/utils.js]: brackets

File: src/components/SlateEditor/plugins/detailsbox/DetailsBox.jsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ const DetailsBox = props => {
   };
 
   return (
-    <div css={detailsWrapper} {...props.attributes}>
+    <div css={detailsWrapper} draggable={!showEditModal} {...props.attributes}>
```
Comment (chrpeter) [inline src/components/SlateEditor/plugins/detailsbox/DetailsBox.jsx]: hvorfor ikke styled? Vet det er gammel kode, men fikse opp?

File: src/components/SlateEditor/plugins/DND/onDrop.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    !topLevelTarget ||
+    topLevelTarget.key === topLevelSource.key
+  ) {
+    return next();
+  }
+
+  const { key: topLevelSourceKey } = topLevelSource;
+
+  if (topLevelTarget.type === 'paragraph' && topLevelTarget.text === '') {
+    editor.withoutNormalizing(() => {
```
Comment (chrpeter) [inline src/components/SlateEditor/plugins/DND/onDrop.js]: Burde det ikke normaliseres når det er en bruker som gjør noe? Feks flytter et bilde så vil vi jo at det skal normaliseres?

File: src/components/SlateEditor/plugins/detailsbox/DetailsBox.jsx (hunk 3 above)
Comment (gunnarvelle) [reply src/components/SlateEditor/plugins/detailsbox/DetailsBox.jsx]: Kan ta dette seinare.

**Pull request merged**